File: server/internal/domain/news.go
package domain

type News struct {
	ID       string   `json:"id"`
	DateTime string   `json:"datetime"`
	TitleRu  string   `json:"title_ru"`
	TitleEn  string   `json:"title_en"`
	Dir      string   `json:"dir"`
	MainImage  string   `json:"main_image"`
	TextRu   string   `json:"text_ru"`
	TextEn   string   `json:"text_en"`
	Images   []string `json:"images,omitempty"`
	Videos   []string `json:"videos,omitempty"`
}

type NewsFilter struct {
	Status string
	Query  string
	Page   int
	Limit  int
}

File: client/app/api/news.ts
import { getHttpClient, } from '~/api/http-client';

export interface NewsItem {
    id: string
    datetime: string
    title_ru: string
    title_en: string
    dir: string
    main_image: string
    text_ru: string
    text_en: string
    images: string[]
    videos: string[]
}

export interface NewsListResponse {
    news: NewsItem[]
    total: number
}

export async function fetchNews(params?: {
    status?: string
    page?: number
    limit?: number
},): Promise<NewsListResponse> {
    try {
        const { data, } = await getHttpClient().get<NewsListResponse>('news', { params, },);
        return data;
    } catch (e) {
        console.error('fetchNews error:', e,);
        return { news: [], total: 0, };
    }
}

export async function fetchNewsById(id: string,): Promise<NewsItem | null> {
    try {
        const { data, } = await getHttpClient().get<NewsItem>(`news/${id}`,);
        return data;
    } catch (e) {
        console.error('fetchNewsById error:', e,);
        return null;
    }
}
